sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -471,38 +469,93 @@ private static void buildAndSend(HttpRequest httpRequest, OutputStreamWriter out
          *
          * @param httpRequest The HTTP Request being sent
-         * @param reader the input stream from the socket
-         * @return an instance of HttpUrlConnectionResponse
+         * @param inputStream the input stream from the socket
+         * @return an instance of Response
          * @throws IOException If an I/O error occurs
          */
-        private static Response<?> buildResponse(HttpRequest httpRequest, BufferedReader reader) throws IOException {
-            String statusLine = reader.readLine();
-            int dotIndex = statusLine.indexOf('.');
-            int statusCode = Integer.parseInt(statusLine.substring(dotIndex + 3, dotIndex + 6));
+        private static Response<?> buildResponse(HttpRequest httpRequest, BufferedInputStream inputStream)
+            throws IOException {
+            // Parse Http response from socket:
+            // Status Line
+            // Response Headers
+            // Blank Line
+            // Response Body
+
+            int statusCode = readStatusCode(inputStream);
+            HttpHeaders headers = readResponseHeaders(inputStream);
+            // read body if present
+            // TODO: (add chunked encoding support)
+            HttpHeader contentLengthHeader = headers.get(CONTENT_LENGTH);
+            byte[] body = getBody(inputStream, contentLengthHeader);
+            if (body != null) {
+                return new HttpResponse<>(httpRequest, statusCode, headers, BinaryData.fromBytes(body));
+            }
+            return new HttpResponse<>(httpRequest, statusCode, headers, null);
+        }
 
-            HttpHeaders headers = new HttpHeaders();
-            String line;
-            while ((line = reader.readLine()) != null && !line.isEmpty()) {
-                // Headers may have optional leading and trailing whitespace around the header value.
-                // https://tools.ietf.org/html/rfc7230#section-3.2
-                // Process this accordingly.
-                int split = line.indexOf(':'); // Find ':' to split the header name and value.
-                String key = line.substring(0, split); // Get the header name.
-                String value = line.substring(split + 1).trim(); // Get the header value and trim whitespace.
-                headers.add(HttpHeaderName.fromString(key), value);
+        private static byte[] getBody(BufferedInputStream inputStream, HttpHeader contentLengthHeader)
+            throws IOException {
+            int contentLength;
+            if (contentLengthHeader == null || contentLengthHeader.getValue() == null) {
+                contentLength = -1;
+            } else {
+                contentLength = Integer.parseInt(contentLengthHeader.getValue());
             }
+            if (contentLength > 0) {
+                byte[] buffer = new byte[contentLength];
+                int bytesRead;
+                int totalBytesRead = 0;
+
+                while (totalBytesRead < contentLength
```

Comment:
This could be a problem if there is an invalid response where the amount of data differs from the Content-Length value. We probably should just read all the data and then if the read amount differs from what the Content-Length header specified throw an exception.

Reply:
Should we rather throw when the "amount of data differs from the Content-Length value" instead?

Reply:
Yeah, but I'd do it after completing the reading